### PR TITLE
Allow for passing through of MF2 from upstream

### PR DESCRIPTION
**Pull request by dshanske**
This allows for parsed MF2 json to be passed through. As the goal is to move MF2 parsing into webmentions, this adds an intermediate step of not parsing twice as we move more of the code over.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Semantic-Linkbacks #
-**Contributors:** [pfefferle](https://profiles.wordpress.org/pfefferle), [dshanske](https://profiles.wordpress.org/dshanske)  
+**Contributors:** pfefferle, dshanske
```
Comment (pfefferle) [inline README.md]: Can you please update run „npm update“ and re-run grunt?

**Comment by pfefferle** [comment]
I would prefer the other way... I think it is simpler to start loading the mf2 Lob in the Webmention Plugin and use it here. Then it deals simpler to move the parsing step by step.

**Comment by dshanske** [comment]
> I would prefer the other way... I think it is simpler to start loading the mf2 Lob in the Webmention Plugin and use it here. Then it deals simpler to move the parsing step by step.

I'm confused. Isn't that what I tried to do here? Passing the mf2 through from webmentions? But if it isn't passed through, then Semantic Linkbacks will try again to get it. That covers people who haven't upgraded yet, and in the next release, we can remove it completely from this plugin.

**Comment by pfefferle** [comment]
Ah, it was the mf2 License, sorry 😞

**Comment by dshanske** [comment]
I think I fixed the issue you requested

**Comment by dshanske** [comment]
@pfefferle Can we merge and push this one? So that we can start parsing mf2 on the webmention side?

**Comment by dshanske** [comment]
@pfefferle Still confused about your comment here

File: includes/class-linkbacks-mf2-handler.php [hunk 2]
```diff
@@ -139,13 +139,17 @@ public static function generate_commentdata( $commentdata ) {
...
-		// parse source html
-		$parser   = new Parser( $commentdata['remote_source_original'], $source );
-		$mf_array = $parser->parse( true );
+		if ( ! array_key_exists( 'remote_source_mf2', $commentdata ) ) {
+			if ( ! class_exists( 'Mf2\Parser' ) ) {
+				require_once plugin_dir_path( __DIR__ ) . 'vendor/mf2/mf2/Mf2/Parser.php';
+			}
+
+			// parse source html
+			$parser                           = new Parser( $commentdata['remote_source_original'], $source );
```
Comment (pfefferle) [inline includes/class-linkbacks-mf2-handler.php:149]: I think this does not work any more, because Webmention is using a different namespace.

Comment (dshanske) [reply includes/class-linkbacks-mf2-handler.php:149]: You are right, forgot about that

Comment (dshanske) [reply includes/class-linkbacks-mf2-handler.php:149]: But this is Semantic Linkbacks... which still bundles the code

**Pull request closed without merging**